include/VX/vx_khr_safe_casts.h

```diff
@@ -71,6 +71,10 @@ __MAKE_SAFE_CASTS__(threshold, Threshold)
 #ifdef VX_TYPE_USER_DATA_OBJECT
 __MAKE_SAFE_CASTS__(user_data_object, UserDataObject)
 #endif
+#if defined(LINUX) || defined(QNX)
```

Comment:
why this protection? why don't we create a #ifdef VX_PRODUCER_CONSUMER_EXT define as #ifdef VX_TYPE_USER_DATA_OBJECT?